### PR TITLE
add minimum in mathematical function 

### LINKED ISSUE #3715: minimum
Add minimum to NumPy frontend one of the task of #1525

### PR DESCRIPTION
**Pull request by AmeenUrRehman**
Add minimum mathematical function one of the tasks of #1525  

resolves #3715

### CODE REVIEW

**Comment by AmeenUrRehman** [comment]
@JLU-Neal Could we ready to merge plz check

**Comment by AmeenUrRehman** [comment]
@juliagsy hey could review my code plz

**Comment by AmeenUrRehman** [comment]
@VedPatwardhan Hey Could you review my code plz.

**Comment by AmeenUrRehman** [comment]
Issue no. #3691

**Comment by AmeenUrRehman** [comment]
@JLU-Neal could you plz check my code plz

**Comment by AmeenUrRehman** [comment]
@Daniel4078 hey could you plz check my code Thank You!

**Review by AmeenUrRehman — COMMENTED** [review]
@JLU-Neal Plz check my code

**Comment by AmeenUrRehman** [comment]
@juliagsy Could you plz review my code and check if we merge this

**Comment by juliagsy** [comment]
Hey there! ahh sure, I'll review soon, sorry for the delay!

File: ivy/functional/frontends/numpy/mathematical_functions/extrema_finding.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import ivy
+
+# local
+from ivy.func_wrapper import from_zero_dim_arrays_to_float
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/mathematical_functions/extrema_finding.py:4]: seems like unused import, please remove if it's not used

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_extrema_finding.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+# minimum
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=ivy_np.valid_float_dtypes, num_arrays=2
+    ),
+    dtype=st.sampled_from(ivy_np.valid_float_dtypes + (None,)),
+    where=np_frontend_helpers.where(),
+    as_variable=helpers.array_bools(),
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_extrema_finding.py:19]: please add the `num_arrays=2` parameters in data generation for `as_variable` and `native_array` as there are 2 arrays involved in the operation

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_extrema_finding.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    with_out=st.booleans(),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.numpy.minimum"
+    ),
+    native_array=helpers.array_bools()
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_extrema_finding.py:24]: above comment

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_extrema_finding.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+# minimum
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=ivy_np.valid_float_dtypes, num_arrays=2
+    ),
+    dtype=st.sampled_from(ivy_np.valid_float_dtypes + (None,)),
+    where=np_frontend_helpers.where(),
+    as_variable=helpers.array_bools(),
+    with_out=st.booleans(),
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_extrema_finding.py:20]: you can remove this as it is handled in the `handle_cmd_line_arg` decorator, but please keep it in the` test_numpy_minimum` argument list

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey there! Thanks for contributing! There are slight changes needed and you're almost good to go! As usual, please request my review once you're done! Thank you!

**Comment by AmeenUrRehman** [comment]
@juliagsy Plz review this I've try to resolve all issues

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_extrema_finding.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import ivy.functional.backends.numpy as ivy_np
+import ivy_tests.test_ivy.test_frontends.test_numpy.helpers as np_frontend_helpers
+from ivy_tests.test_ivy.helpers import handle_cmd_line_args
+
+
+# minimum
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=ivy_np.valid_float_dtypes,
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_extrema_finding.py:16]: please also add a parameter `num_arrays=2` in this helper functions

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hello! Great work so far! One last small fix and you're almost good to go! Please request my review again once you're done! Thanks a lot!

**Comment by AmeenUrRehman** [comment]
@juliagsy Thank You! for your guidance plz review the code I've fixes the issue

**Review by juliagsy — APPROVED** [review]
Thanks for contributing! I will merge the PR soon!

**Pull request merged**